### PR TITLE
Robolectric unit testing

### PR DESCRIPTION
**Pull request by daboome**
Added one test method to test simple circle focus. The test method specifies view position settings (margin left, margin top) to focus on, as well as view settings (view width, view height). Also circle radius factor is being set. Test method creates Activity, inflates it with layout containing specified view to focus on, makes calculations and asserts results. FancyImageView's fields are intentionally made package local for the sake of unit testing.
This is simple state-test. Please take a look, and tell if it's ok, so i can apply the same logic to other cases.

### CODE REVIEW

File: library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java [hunk 1]
```diff
@@ -14,19 +14,19 @@
...
+    int mBackgroundColor = Color.TRANSPARENT;
+    int mFocusBorderColor = Color.TRANSPARENT;
+    int mFocusBorderSize;
+    int mRoundRectRadius = 20;
+    Calculator mCalculator;
+    int mAnimCounter = 0;
+    int mStep = 1;
+    double mAnimMoveFactor = 1;
+    Path mPath;
+    RectF rectF;
```
Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: Do you need to make all these fields package local?

Comment (daboome) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: Yep, as I said this is made for the state testing. Anyway fields are not visible outside library. Library you provided with the link in issue #9 has all fields package local visibile too.

Comment (daboome) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: another solution would be adding getters for the corresponding fields.

Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: Ok I understand. This is `FancyImageView`. I thought that it is `FancyShowCaseView` :)

Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: I think this approach is ok.

Comment (daboome) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: ok, then i'm gonna add test methods to other cases

Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: Can you wait for me? I will pull and test once. If it is ok you can go on.

Comment (daboome) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: ok, waiting for your response

Comment (daboome) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: did you complete testing?

File: library/src/test/java/me/toptas/fancyshowcase/ExampleUnitTest.java [hunk 2]
```diff
@@ -1,17 +1,82 @@
...
+        int [] points = new int[2];
+        button.getLocationInWindow(points);
+        int mCircleCenterX = points[0] + viewWidth / 2;
+        int mCircleCenterY = points[1] + viewHeight / 2 - fancyImageView.mCalculator.adjustHeight(activity, fitSystemWindows);
+        int radius = (int) ((int) (Math.hypot(button.getWidth(), button.getHeight()) / 2) * circleRadiusFactor);
+        // Asserting phase
+        Assert.assertEquals(FocusShape.CIRCLE, fancyImageView.mCalculator.getFocusShape());
+        Assert.assertEquals(mCircleCenterX, fancyImageView.mCalculator.getCircleCenterX());
+        Assert.assertEquals(mCircleCenterY, fancyImageView.mCalculator.getCircleCenterY());
+        Assert.assertEquals(radius, fancyImageView.mCalculator.getViewRadius());
```
Comment (faruktoptas) [inline library/src/test/java/me/toptas/fancyshowcase/ExampleUnitTest.java]: Assertions only check calculated values not the real drawn view's values.

File: library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java (hunk 1 above)
Comment (faruktoptas) [inline library/src/main/java/me/toptas/fancyshowcase/FancyImageView.java]: Sorry for late response. I wrote a comment.

Assertions only check calculated values not the real drawn view's values.

File: library/src/test/java/me/toptas/fancyshowcase/ExampleUnitTest.java (hunk 2 above)
Comment (faruktoptas) [inline library/src/test/java/me/toptas/fancyshowcase/ExampleUnitTest.java]: I deeply checked and debugged again. This approach ok for me. There is only one point I want to say: 
`fitSystemWindows` should be tested in a different way. In this unit test, setting `fitSystemWindows` to true or false does not matter. Both passes. But changing this parameter changes the focus point.

**Comment by faruktoptas** [comment]
Are  you still working on this issue?

**Comment by daboome** [comment]
The Calculator should be refactored in some way, because fitSystemWindows affects calculating mCircleCenterY. and it should be passed as a param. I think there should be a way of handling fitSystemWindows param so that it wouldn't be acting in calculations. I'll try to make some changes in code a bit later.

**Comment by faruktoptas** [comment]
You are right. Calculator should be refactored to test fitSystemWindows parameter.

**Comment by faruktoptas** [comment]
Closing this PR because JUnit and Espresso tests are being used instead of Robolectric.

**Pull request closed without merging**